src/nl/rubensten/texifyidea/action/tablewizard/LatexTableWizardAction.kt

```diff
@@ -0,0 +1,124 @@
+        val editors = FileEditorManager.getInstance(project!!).selectedEditors
+        val editor = editors.filter { it is TextEditor }.map { it as TextEditor }.first()
+        val document = editor.editor.document
+
+        // Get the indentation from the current line.
+        val indent = document.lineIndentation(document.getLineNumber(editor.editor.caretOffset()))
+
+        // Create the dialog.
+        val dialogWrapper = TableCreationDialogWrapper()
+        // If the user pressed OK, do stuff.
+        if (dialogWrapper.showAndGet()) {
+
+            // Get the table information from the dialog, and convert it to latex.
+            val tableTextToInsert = convertTableToLatex(dialogWrapper.tableInformation, indent)
+
+            // Use an insert action to insert the table.
+            InsertTable(tableTextToInsert).actionPerformed(file, project, editor)
+
+            // Insert the booktabs package.
+            WriteCommandAction.runWriteCommandAction(project,
+                    "Insert table",
+                    "LaTeX",
+                    Runnable { file!!.psiFile(project)!!.insertUsepackage(Package.BOOKTABS) },
+                    file!!.psiFile(project)
+            )
+        }
+
+    }
+
+    /**
+     * Convert the table information to a latex table that can be inserted into the file.
+     *
+     * @param tableInformation
+     * @param lineIndent is the indentation of the current line, to be used on each new line.
+     * @param tabIndent is the continuation indent.
+     */
+    private fun convertTableToLatex(tableInformation: TableInformation, lineIndent: String, tabIndent: String = "    "): String {
+        /**
+         * Local function to process the contents of the table, i.e. the header text and table entries. The indentation
+         * is the same throughout the table contents.
+         */
+        fun processTableContent(indent: String): String {
+            return with(tableInformation) {
+                val headers = tableModel.getColumnNames()
+                        .joinToString(
+                                prefix = "$indent\\toprule\n$indent",
+                                separator = " & ",
+                                postfix = " \\\\\n$indent\\midrule\n"
+                        ) { "\\textbf{$it}" }
+
+                val rows = tableModel.dataVector.joinToString(separator = "\n", postfix = "\n$indent\\bottomrule\n") { row ->
+                    (row as Vector<*>).joinToString(
+                            prefix = indent,
+                            separator = " & ",
+                            postfix = " \\\\"
+                    ) {
+                        // Enclose with $ if the type of this column is math.
+                        val index = row.indexOf(it)
+                        val encloseWith = if (columnTypes[index] == ColumnType.MATH_COLUMN) "$" else ""
+                        encloseWith + it.toString() + encloseWith
```

Comment:
I get an NPE here when I attempt to create a table as follows: two columns (text and numbers) with a single empty row.

<details>
<summary>Stacktrace</summary>

```
java.lang.NullPointerException
	at nl.rubensten.texifyidea.action.tablewizard.LatexTableWizardAction$convertTableToLatex$1$processTableContent$$inlined$with$lambda$1$1.invoke(LatexTableWizardAction.kt:85)
	at nl.rubensten.texifyidea.action.tablewizard.LatexTableWizardAction$convertTableToLatex$1$processTableContent$$inlined$with$lambda$1$1.invoke(LatexTableWizardAction.kt:22)
	at kotlin.text.StringsKt__StringBuilderKt.appendElement(StringBuilder.kt:58)
	at kotlin.collections.CollectionsKt___CollectionsKt.joinTo(_Collections.kt:2291)
	at kotlin.collections.CollectionsKt___CollectionsKt.joinToString(_Collections.kt:2308)
	at kotlin.collections.CollectionsKt___CollectionsKt.joinToString$defa	ult(_Collections.kt:2307)
	at nl.rubensten.texifyidea.action.tablewizard.LatexTableWizardAction$convertTableToLatex$1$processTableContent$$inlined$with$lambda$1.invoke(LatexTableWizardAction.kt:77)
	at nl.rubensten.texifyidea.action.tablewizard.LatexTableWizardAction$convertTableToLatex$1$processTableContent$$inlined$with$lambda$1.invoke(LatexTableWizardAction.kt:22)
	at kotlin.text.StringsKt__StringBuilderKt.appendElement(StringBuilder.kt:58)
	at kotlin.collections.CollectionsKt___CollectionsKt.joinTo(_Collections.kt:2291)
	at kotlin.collections.CollectionsKt___CollectionsKt.joinToString(_Collections.kt:2308)
	at kotlin.collections.CollectionsKt___CollectionsKt.joinToString$default(_Collections.kt:2307)
	at nl.rubensten.texifyidea.action.tablewizard.LatexTableWizardAction$convertTableToLatex$1.invoke(LatexTableWizardAction.kt:76)
	at nl.rubensten.texifyidea.action.tablewizard.LatexTableWizardAction.convertTableToLatex(LatexTableWizardAction.kt:101)
	at nl.rubensten.texifyidea.action.tablewizard.LatexTableWizardAction.convertTableToLatex$default(LatexTableWizardAction.kt:62)
	at nl.rubensten.texifyidea.action.tablewizard.LatexTableWizardAction.actionPerformed(LatexTableWizardAction.kt:39)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:260)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:277)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.lambda$actionPerformed$0(ActionMenuItem.java:292)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.runOnOwnContext(FocusManagerImpl.java:283)
	at com.intellij.openapi.wm.impl.IdeFocusManagerImpl.runOnOwnContext(IdeFocusManagerImpl.java:107)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.actionPerformed(ActionMenuItem.java:282)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem.lambda$fireActionPerformed$0(ActionMenuItem.java:111)
	at com.intellij.openapi.application.TransactionGuardImpl.runSyncTransaction(TransactionGuardImpl.java:88)
	at com.intellij.openapi.application.TransactionGuardImpl.lambda$submitTransaction$1(TransactionGuardImpl.java:111)
	at com.intellij.openapi.application.TransactionGuardImpl.submitTransaction(TransactionGuardImpl.java:120)
	at com.intellij.openapi.application.TransactionGuard.submitTransaction(TransactionGuard.java:121)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem.fireActionPerformed(ActionMenuItem.java:111)
	at com.intellij.ui.plaf.beg.BegMenuItemUI.doClick(BegMenuItemUI.java:524)
	at com.intellij.ui.plaf.beg.BegMenuItemUI.access$300(BegMenuItemUI.java:35)
	at com.intellij.ui.plaf.beg.BegMenuItemUI$MyMouseInputHandler.mouseReleased(BegMenuItemUI.java:546)
	at java.awt.Component.processMouseEvent(Component.java:6550)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3325)
	at java.awt.Component.processEvent(Component.java:6315)
	at java.awt.Container.processEvent(Container.java:2239)
	at java.awt.Component.dispatchEventImpl(Component.java:4899)
	at java.awt.Container.dispatchEventImpl(Container.java:2297)
	at java.awt.Component.dispatchEvent(Component.java:4721)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)
	at java.awt.Container.dispatchEventImpl(Container.java:2283)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4721)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:766)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
	at java.awt.EventQueue$4.run(EventQueue.java:739)
	at java.awt.EventQueue$4.run(EventQueue.java:737)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:736)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:747)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:692)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:391)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

</details>

Reply:
This should be fixed now. When adding a column it would set all cells in the new column to null (instead of the empty string).